lang/compiler.go

```diff
@@ -428,18 +428,31 @@ func (p *AstProgram) getSpec(ctx context.Context, alloc *memory.Allocator) (*flu
 
 	s, cctx := opentracing.StartSpanFromContext(ctx, "eval")
 
-	sideEffects, scope, err := p.Runtime.Eval(cctx, ast, &ExecOptsConfig{}, flux.SetNowOption(p.Now))
+	// Set the now option to our own default and capture the option itself
+	// to allow us to find it after the run. A user might overwrite the
+	// now parameter with their own thing so we don't want to allow for
+	// that interference. If `option now` is used to overwrite this,
+	// the inner value pointed to by the option will be modified.
+	// TODO(jsternberg): Personal note, I don't like how now interacts with
+	// the runtime and flux code in so many places. We should evaluate how
+	// now is used and see if we can improve how now interacts with the system.
```

Comment:
Agree it needs some cleanup. Since the `tableFind` functions need to have now available in the execution dependencies, I think that should become the authority on `now`.